.github/workflows/update_package.yml

```diff
@@ -25,24 +25,31 @@ jobs:
           New-Item test_logs -itemType Directory | Out-Null
           foreach ($packagePath in (Get-ChildItem packages)){
             $package = $packagePath.Name
-            $version = python scripts\utils\update_package.py $package
-            $updated = $?
-            echo "$package $version"
-            if ($updated -and $version) {
-              # Test package before committing
-              scripts\test\test_install.ps1 $package *>> "test_logs/$package.txt"
-              $tested = $?
-              cd $root
-              if ($tested) {
-                git add "packages/$package/"
-                git commit -m "Update $package to $version"
-              } else {
-                echo "$package $version FAILED"
+            $newVersion = 0
+            # Test indepdendly every type of update and commit what works
+            foreach ($UPDATE_TYPE in ('DEPENDENCIES', 'GITHUB_URL')) {
+              $version = python scripts\utils\update_package.py $package --update_type $UPDATE_TYPE
```

Comment:
For a potential follow up, would it make sense to turn this into a Python script or add to `update_package`?

Reply:
not sure I understand what you mean. There is already a way to run `update_package.py` with both update types (note the definition of `ALL = DEPENDENCIES | GITHUB_URL`). But we need to run them separately here to ensure we can commit only the part which passes the tests.

Reply:
Sorry, I mean to do the surrounding code in Python instead of PS.

Reply:
We could rewrite the surrounding code in Python but not sure if we win much. We still need to call the test powershell script. 🤔 Let's merge this (as this is merging the linter of categories) and continue the discussion afterwards.